src/Router/buildServerApp.tsx

```diff
@@ -44,17 +53,29 @@ export function buildServerApp(config: AppConfig): Promise<ServerResolveProps> {
         return
       }
 
-      const bootProps = {
-        system: { ...config, relayEnvironment, resolver, routes, currentUser },
-      }
+      const system: Router = { relayEnvironment, resolver, routes, currentUser }
 
-      const AppContainer = props => {
+      const AppContainer: React.SFC<AppShellProps> = props => {
         return (
-          <Boot {...bootProps} {...props}>
-            <AppShell
-              data={props.relayData}
-              loadableState={props.loadableState}
-            >
+          <Boot
+            system={system}
+            initialBreakpoint={initialBreakpoint}
+            initialState={[
+              new AppState({ ...initialAppState, system }),
+              ...initialState,
+            ]}
+          >
+            {config.subscribe &&
+              config.subscribe.to &&
+              config.subscribe.onChange && (
+                <Subscribe to={config.subscribe.to}>
+                  {(...args) => {
+                    config.subscribe.onChange(...args)
+                    return null
+                  }}
+                </Subscribe>
+              )}
+            <AppShell data={props.data} loadableState={props.loadableState}>
```

Comment:
The `data` was being passed as `props.relayData`, which doesn’t exist https://github.com/artsy/reaction/pull/1003/files#diff-078cfdc29c36b8e0e6e77f70fba6ae7bR89. This was leading to no data being hydrated for the client side and thus on the client side an additional identical fetch was being made.